editor/src/components/canvas/ui-jsx-canvas-renderer/ui-jsx-canvas-component-renderer.tsx

```diff
@@ -90,23 +92,40 @@ export function createComponentRendererComponent(params: {
 
     const mutableContext = params.mutableContextRef.current[params.filePath].mutableContext
 
-    const { topLevelElements, imports } = useGetTopLevelElementsAndImports(params.filePath)
-    const { code, highlightBounds } = useGetCodeAndHighlightBounds(params.filePath)
+    const shouldUpdate = () =>
+      !isFeatureEnabled('Canvas Selective Rerender') ||
+      ElementsToRerenderGLOBAL.current === 'rerender-all-elements' ||
+      ElementsToRerenderGLOBAL.current.findIndex((er) => {
```

Comment:
Wouldn't `ElementsToRerenderGLOBAL.current.some(...` make more sense?

Reply:
fixed in 1fd1b4c